main.tf

```diff
@@ -102,7 +102,10 @@ locals {
       {
         "project_root"        = replace(format("%s/%s", var.root_modules_path, module), "../", "")
         "root_module"         = module,
-        "terraform_workspace" = trimsuffix(file, ".yaml"),
+        "terraform_workspace" = try(content.default_tf_workspace_enabled, var.default_tf_workspace_enabled) ? "default" : trimsuffix(file, ".yaml"),
+        # `yaml` is intentionally used here as we require Stack and `tfvars` config files to be named equally
+        # TODO: Add tests to ensure that the `tfvars` file is named the same as the Stack config file
```

Comment:
This is a good TODO -- One that we should follow up on. I wonder if we can make this a pre-condition?

Reply:
Hmm, the precondition seems reasonable here; I added one.
Since we operate a lot with stack yaml file content, we don't need to check if it exists, but we should check if a related tfvars file is present. This is crucial for multi-instance model (ha, we're already using this!).

We can add a check to ensure that there is a stack config to each tfvars file. This would throw a warning and won't hault TF execution in case this is expected. 
How does that sound to you?

Reply:
I think the pre-condition looks great now that I see it / think about it more -- Damn check blocks are confusing.

Let's stick with what you've got! Going to merge so we can share with client 🤝